### PR TITLE
add direction to CupertinoPickerDefaultSelectionOverlay

### PR DESCRIPTION
**Pull request by Dan-Crane**
Added direction support to `CupertinoPickerDefaultSelectionOverlay`. When direction in app is RTL, "selection overlay" in CupertinoDatePicker does not work correctly.

<details>
<summary>Bug image</summary>
<br>

Simulator Screen Shot - iPhone 13 - 2021-11-03 at 14 07 46

</details>

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Dan-Crane** [comment]
@googlebot I signed it!

**Comment by LongCatIsLooong** [comment]
Thanks for contributing! Could you add a test to verify it now works correctly with RTL text?

**Comment by Hixie** [comment]
@Dan-Crane Just checking in, are you still interested in working on this PR? I'm afraid we can't accept PRs that don't have tests, but we'd be happy to advise on how to write tests if you're interested (see our contributor docs for a link to our Discord where you can get help).

**Comment by Dan-Crane** [comment]
Oh! Sorry, i didn't notice. Yes, I will try to write tests. Thanks

File: packages/flutter/lib/src/cupertino/picker.dart [hunk 1]
```diff
@@ -379,14 +379,14 @@ class CupertinoPickerDefaultSelectionOverlay extends StatelessWidget {
     const Radius radius = Radius.circular(_defaultSelectionOverlayRadius);
 
     return Container(
-      margin: EdgeInsets.only(
-        left: capLeftEdge ? _defaultSelectionOverlayHorizontalMargin : 0,
-        right: capRightEdge ? _defaultSelectionOverlayHorizontalMargin : 0,
+      margin: EdgeInsetsDirectional.only(
+        start: capLeftEdge ? _defaultSelectionOverlayHorizontalMargin : 0,
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/picker.dart:383]: I would recommend renaming the parameter to capStartEdge or capLeadingEdge.

File: packages/flutter/test/cupertino/picker_test.dart [hunk 2]
```diff
@@ -422,4 +422,28 @@ void main() {
       );
     }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.macOS }));
   });
+
+  group('CupertinoPickerDefaultSelectionOverlay', () {
+    testWidgets('should be used directional decoration', (WidgetTester tester) async {
```
Comment (LongCatIsLooong) [inline packages/flutter/test/cupertino/picker_test.dart:427]: nit: used -> using?

File: packages/flutter/lib/src/cupertino/picker.dart (hunk 1 above)
Comment (Dan-Crane) [reply packages/flutter/lib/src/cupertino/picker.dart]: Thanks, you are right. 
done

File: packages/flutter/test/cupertino/picker_test.dart (hunk 2 above)
Comment (Dan-Crane) [reply packages/flutter/test/cupertino/picker_test.dart]: done

**Review by LongCatIsLooong — APPROVED** [review]
LGTM. Thanks for contributing!

**Review by jmagman — APPROVED** [review]
LGTM!

**Pull request merged**